scripts/nbd-info.nse

```diff
@@ -0,0 +1,189 @@
+  -- Run the LIST command and store the responses.
+  local req = comm:build_opt_req("LIST")
+  if not req then
+    return
+  end
+
+  local status, err = comm:send(req)
+  if not status then
+    stdnse.debug1("Failed to send option request: %s", err)
+    return nil
+  end
+
+  while true do
+    local rep = comm:receive_opt_rep()
+    if not rep or rep.rtype_name ~= "SERVER" then
+      break
+    end
+
+    comm.exports[rep.export_name] = {}
+  end
+end
+
+newstyle_connection = function(comm, args)
+  local names = {}
+
+  for _, name in ipairs(args.export_name) do
+    table.insert(names, name)
+  end
+
+  for name, _ in pairs(comm.exports) do
+    table.insert(names, name)
+  end
+
+  for i, name in ipairs(names) do
+    if i ~= 1 then
+      local status = comm:reconnect()
+      if not status then
+	return
+      end
+    end
+
+    comm:attach(name)
+  end
+end
+
+local function parse_args()
+  local args = {}
+
+  local arg = stdnse.get_script_args(SCRIPT_NAME .. ".export-names")
+  if not arg then
+    -- An empty string for an export name indicates to the server that
+    -- we wish to attach to the default export.
+    arg = {}
+  elseif type(arg) ~= table then
+    arg = {arg}
+  end
+  args.export_name = arg
+
+  return args
+end
+
+action = function(host, port)
+  local args = parse_args()
+
+  local comm = nbd.Comm:new(host, port)
+
+  local status = comm:connect(args)
+  if not status then
+     return nil
+  end
+
+  -- If the service supports an unrecognized negotiation, or the
+  -- oldstyle negotiation, there's no more information to be had.
+  if comm.protocol.negotiation == "unrecognized" or comm.protocol.negotiation == "oldstyle" then
+    -- Nothing to do.
+    comm:close()
+
+  -- If the service supports the (non-fixed) newstyle negotiation,
+  -- which should be very rare, we can only send a single option. That
+  -- option is the name of the export to which we'd like to attach.
+  elseif comm.protocol.negotiation == "newstyle" then
+    newstyle_connection(comm, args)
+
+  -- If the service supports the fixed newstyle negotiation, then we
+  -- can perform option haggling to wring additional information from
+  -- it.
+  elseif comm.protocol.negotiation == "fixed newstyle" then
+    enumerate_options(comm)
+    newstyle_connection(comm, args)
+
+  -- Otherwise, we've got a mismatch between the library and this script.
+  else
+    assert(false, "NBD library supports more negotiation styles than this script.")
+  end
+
+  -- Master output table.
+  local output = stdnse.output_table()
+
+  -- Format protocol information.
+  local tbl = {}
+  if comm.protocol.negotiation == "oldstyle" and comm.exports["(default)"] then
+    if comm.exports["(default)"].hflags & nbd.NBD.handshake_flags.FIXED_NEWSTYLE then
+      table.insert(tbl, "The server software appears to support fixed newstyle negotiation, but not on this port.")
+    end
+  end
+  table.insert(tbl, ("Negotiation: %s"):format(comm.protocol.negotiation))
```

Comment:
Instead of inserting strings containing "key: value", make `tbl` a `stdnse.output_table` and set `tbl[key] = value`. I know it doesn't work with the  "not on this port" thing, but maybe if you had a key like "newstyle negotiation supported:" and the value is "other port"? The other thing to try would be to use both integer (with `table.insert`) and string keys. I know NSE can handle that, but I don't know if it works via `output_table`.